legal-api/src/legal_api/reports/report.py

```diff
@@ -245,18 +191,85 @@ def _get_template_data(self):  # pylint: disable=too-many-branches
                 filing['effective_date'] = agm_date.strftime('%B %d, %Y')
             else:
                 filing['agm_date'] = 'No AGM'
-
         elif self._filing.filing_type in ('changeOfAddress', 'changeOfDirectors'):
             # for standalone filings, the effective date comes from the filing data
             filing['effective_date'] = effective_date.strftime('%B %d, %Y')
 
+    def _set_directors(self, filing):
+        if filing.get('changeOfDirectors'):
+            filing['listOfDirectors'] = filing['changeOfDirectors']
+        else:
+            filing['listOfDirectors'] = {
+                'directors': filing['annualReport']['directors']
+            }
+        # create helper lists of appointed and ceased directors
+        directors = self._format_directors(filing['listOfDirectors']['directors'])
+        filing['listOfDirectors']['directorsAppointed'] = [el for el in directors if 'appointed' in el['actions']]
+        filing['listOfDirectors']['directorsCeased'] = [el for el in directors if 'ceased' in el['actions']]
+
+    def _format_directors(self, directors):
+        for director in directors:
+            try:
+                self._format_address(director['deliveryAddress'])
+                self._format_address(director['mailingAddress'])
+            except KeyError:
+                pass
+        return directors
+
+    def _set_addresses(self, filing):
```

Comment:
Does this code need to handle the "recordsOffice" addresses (ie, for IA office or IA parties)?

Reply:
This section of code is for the cooperatives reports and as far as i know, bcomps dont have reports implemented. Once they need it they should definitely update this section.